client/boot/index.js

```diff
@@ -177,12 +178,14 @@ function boot() {
 			LoggedOutLayout = require( 'layout/logged-out' );
 		}
 
-		layout = React.render(
-			React.createElement( LoggedOutLayout ),
-			document.getElementById( 'wpcom' )
-		);
+		layoutComponentCreator = () => React.createElement( LoggedOutLayout );
 	}
 
+	layout = React.render(
+		React.createElement( ReduxProvider, { store: reduxStore }, layoutComponentCreator ),
```

Comment:
Just a heads-up that I had found in the course of upgrading to React 0.14 in #1498 that this will need to pass the child directly, not as function, even without the upgrade to `react-redux` 4.x.